pennylane_lightning/src/util/Macros.hpp

```diff
@@ -13,12 +13,155 @@
 // limitations under the License.
 /**
```

Comment:
As a follow up, maybe all the macros should not be exposed to the user/dev, but instead maintained here in an anonymous namepsace, and exposed only via module-facing functions that use them. What do you think?

I guess this makes the OMP unrolling a challenge though.

Reply:
As macros and constexpr constants should be available at compile time to use them, they need to be included in the header file, and any variable/definitions in the header files are not really hidden. I am not sure whether current C++ has a solution for this type of data hiding...